### DISCUSSION TITLE
Help understanding why futures::io::copy returns WriteZero error in FRB

### DISCUSSION THREAD

**Discussion by tmpfs** [fzyzcjy/flutter_rust_bridge#1267] [General]
I have some code in a library dependency (which I maintain) that uses `futures::io::copy` to copy data from a reader to a writer.

When I test that code outside of FRB it works as expected however when I call it via FRB the `futures::io::copy` is returning a `WriteZero` (`io::ErrorKind`).

In my API functions I am using `#[tokio::main(flavor = "current_thread")]` as recommended.

I suspect it is something to do with the FRB threadpool but looking for some ideas as to where the problem might be.

Thanks 🙏

**Comment by tmpfs** [comment]
I think this is the error i am hitting:

https://docs.rs/futures-util/0.3.28/src/futures_util/io/copy_buf.rs.html#72

What i fail to understand is why this only happens when using FRB, ideas welcome!

**Reply by fzyzcjy** [reply]
Hmm, firstly, is it only happen with the threadpool crate we are using? If so it can be a bug there.

**Reply by tmpfs** [reply]
@fzyzcjy, I tried to reproduce the problem using `#[tokio::main]` inside a threadpool but unable to reproduce it yet, will keep trying to figure out why this is happening.

**Reply by fzyzcjy** [reply]
Hmm, what if you disable threadpool of frb (by maybe hacking it)?

**Reply by tmpfs** [reply]
@fzyzcjy, I tried also with `worker-single` just in case but that had no effect.

**Reply by tmpfs** [reply]
This is the code (extracted from the library) that is causing the issue when run via FRB https://github.com/tmpfs/threadpool-futures-io-copy but works fine with just `cargo run`.

Next I guess I need to move this into a standalone FRB project to try to reproduce.

**Comment by tmpfs** [comment]
FWIW, if anyone stumbles across this, the bug is no longer present when using FRB v2 🥳
